core/sodasql/common/config_helper.py

```diff
@@ -0,0 +1,125 @@
+
+class ConfigHelper:
+    """Helper class for handling global Soda config.
+    """
+    DEFAULT_CONFIG = {
+        'send_anonymous_usage_stats': True,
+        'user_cookie_id': str(uuid.uuid4())
+    }
+    LOAD_PATHS = ["~/.soda/config.yml", ".soda/config.yml"]
+    __instance = None
+    __config: Dict = {}
+    file_system = FileSystemSingleton.INSTANCE
+
+    @staticmethod
+    def get_instance(path: Optional[str] = None):
+        if ConfigHelper.__instance is None:
+            ConfigHelper()
+        return ConfigHelper.__instance
+
+    def __init__(self, path: Optional[str] = None):
+        if ConfigHelper.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            ConfigHelper.__instance = self
+
+        if path:
+            self.LOAD_PATHS.insert(0, path)
+
+        self.__config = self.config
+
+        if not self.__config:
+            self.init_config_file()
+
+        self.__ensure_basic_config()
+
+    @property
+    def config_path(self) -> str:
+        return self.LOAD_PATHS[0]
+
+    @property
+    def config(self) -> Dict:
+        if not self.__config:
+            self.__config = self.load_config()
+
+        return self.__config
+
+    def load_config(self) -> Dict:
+        config = {}
+        for path in self.LOAD_PATHS:
+            logger.debug(f"Trying to load Soda Config file {path}.")
+
+            if self.file_system.file_exists(path):
+                config = yaml.load(
+                    self.file_system.file_read_as_str(path),
+                    Loader=yaml.SafeLoader
+                )
+                break
+
+        return config
+
+    def get_value(self, key: str, default_value = None):
+        """Get value from loaded config."""
+        return self.config.get(key, default_value)
+
+    def init_config_file(self) -> None:
+        """Init default config file if not present."""
+        destination = self.config_path
+
+        if self.file_system.file_exists(destination):
+            logger.debug(f"Config file {destination} already exists")
+        else:
+            logger.info(f"Creating config YAML file {destination} ...")
+            self.file_system.mkdirs(self.file_system.dirname(destination))
+            self.upsert_config_file(self.DEFAULT_CONFIG)
+
+    def upsert_value(self, key: str, value: str):
+        """Update or insert a value in the config file and refresh loaded state."""
+        config = self.config
+        config[key] = value
+        self.upsert_config_file(config)
+        self.__config = self.load_config()
+
+    def upsert_config_file(self, config: Dict):
+        """Write provided config into a yaml file."""
+        self.file_system.file_write_from_str(
+            self.config_path,
+            yaml.dump(
+                config,
+                default_flow_style=False,
+                sort_keys=False
+            )
+        )
+
+    @classmethod
+    def generate_user_cookie_id() -> str:
```

Comment:
I think Sonar cloud is shouting at this because the `cls` arg is missing

Reply:
actually it can be static as well, changed.